developer/src/kmc-kmn/src/kmw-compiler/validate-layout-file.ts

```diff
@@ -199,6 +199,22 @@ function TransformSpecialKeys14(FDebug: boolean, sLayoutFile: string): string {
   return sLayoutFile;
 }
 
+function TransformSpecialKeys17(FDebug: boolean, sLayoutFile: string): string {
+  // Rewrite Special key labels that are only supported in Keyman 17+
+  // This code is a little ugly but effective.
+  if(!IsKeyboardVersion17OrLater()) {
+    for(let i = 0; i < CSpecialText17Map.length; i++) {
+      // Assumes the JSON output format will not change
+      if(FDebug) {
+        sLayoutFile = sLayoutFile.replaceAll('"text": "'+CSpecialText17Map[i][0]+'"', '"text": this._v>16 ? "'+CSpecialText17Map[i][0]+'" : "'+CSpecialText17Map[i][1]+'"');
+      } else {
+        sLayoutFile = sLayoutFile.replaceAll('"text":"'+CSpecialText17Map[i][0]+'"', '"text":this._v>16?"'+CSpecialText17Map[i][0]+'":"'+CSpecialText17Map[i][1]+'"');
+      }
+    }
```

Comment:
We are using a multidimensional array here instead of a map, due to the conversion from Delphi. One day should clean this up